src/containers/share-sound-to-tiktok-modal/store/sagas.ts

```diff
@@ -0,0 +1,100 @@
+import { takeEvery, put, call, select } from 'redux-saga/effects'
+
+import User from 'models/User'
+import { getAccountUser } from 'store/account/selectors'
+
+import * as actions from './actions'
+import { Status } from './reducer'
+import { getIsAuthenticated } from './selectors'
+
+const TIKTOK_SHARE_SOUND_ENDPOINT =
+  'https://open-api.tiktok.com/share/sound/upload/'
+
+// Because the track blob cannot live in an action (not a POJO),
+// we are creating a singleton here to store it
+let track: Blob | null = null
+
+function* handleShare(action: ReturnType<typeof actions.share>) {
+  yield put(actions.setStatus(Status.SHARE_STARTED))
+
+  // Fetch the track blob
+  const { creator_node_endpoint }: User = yield select(getAccountUser)
+
+  try {
+    const { data } = yield call(
+      window.audiusLibs.File.fetchCID,
```

Comment:
That function only returns the URL to the blob, but I needed the actual data. I could add another parameter to it if you think it's better to avoid calling libs directly